src/main/java/com/github/princesslana/eriscasper/data/util/AuditLogChangeDeserializer.java

```diff
@@ -0,0 +1,143 @@
+package com.github.princesslana.eriscasper.data.util;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.ObjectCodec;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.github.princesslana.eriscasper.data.Snowflake;
+import com.github.princesslana.eriscasper.data.resource.AuditLogChange;
+import com.github.princesslana.eriscasper.data.resource.ImmutableAuditLogChange;
+import com.github.princesslana.eriscasper.data.resource.Overwrite;
+import com.github.princesslana.eriscasper.data.resource.Role;
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.Optional;
+
+public class AuditLogChangeDeserializer extends StdDeserializer<AuditLogChange> {
+
+  public AuditLogChangeDeserializer() {
+    this(null);
+  }
+
+  private AuditLogChangeDeserializer(Class<?> vc) {
```

Comment:
Why is this constructor here?

Reply:
which one?

Reply:
This one that is private. Where is it used? (apart from the other constructor that could call super(null) itself).

Reply:
I'll check but from stuff I saw in documentation it was required for reflective initialization, I'll try to run tests without it though and see how it goes.

Reply:
yeah not needed, I'll remove it. Thought otherwise my bad